plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

```diff
@@ -190,15 +190,8 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
                 .put("timestamptz", handle -> Optional.of(timestampTZColumnMapping(handle.getRequiredDecimalDigits())))
                 .put("date", handle -> Optional.of(ColumnMapping.longMapping(DateType.DATE, (resultSet, columnIndex) -> LocalDate.ofEpochDay(resultSet.getLong(columnIndex)).toEpochDay(), snowFlakeDateWriter())))
                 .put("varchar", handle -> Optional.of(varcharColumnMapping(handle.getRequiredColumnSize(), typeHandle.getCaseSensitivity())))
-                .put("number", handle -> {
-                    int decimalDigits = handle.getRequiredDecimalDigits();
-                    int precision = handle.getRequiredColumnSize() + Math.max(-decimalDigits, 0);
-                    if (precision > 38) {
-                        return Optional.empty();
-                    }
-                    return Optional.of(columnMappingPushdown(
-                            StandardColumnMappings.decimalColumnMapping(DecimalType.createDecimalType(precision, Math.max(decimalDigits, 0)), RoundingMode.UNNECESSARY)));
-                })
+                .put("number", SnowflakeClient::decimalColumnMapping)
+                .put("decimal", SnowflakeClient::decimalColumnMapping)
```

Comment:
Replace `decimal` with `NUMBER` in `toTypeHandle` method instead. 

Also, the method can be renamed to decimalTypeHandle.